awx/main/utils/autoscale.py

```diff
@@ -0,0 +1,29 @@
+import logging
+
+from celery.utils.log import get_logger
+from celery.worker.autoscale import Autoscaler, AUTOSCALE_KEEPALIVE
+from django.conf import settings
+import psutil
+
+logger = get_logger('awx.main.tasks')
+
+
+class DynamicAutoScaler(Autoscaler):
+
+    def __init__(self, pool, max_concurrency, min_concurrency=0, worker=None,
+                 keepalive=AUTOSCALE_KEEPALIVE, mutex=None):
+        super(DynamicAutoScaler, self).__init__(pool, max_concurrency,
+                                                min_concurrency, worker,
+                                                keepalive, mutex)
+        settings_absmem = getattr(settings, 'SYSTEM_TASK_ABS_MEM', None)
+        if settings_absmem is not None:
+            total_memory_gb = int(settings_absmem)
+        else:
+            total_memory_gb = (psutil.virtual_memory().total >> 30) + 1  # noqa: round up
+
+        # 6 workers per GB of total memory
```

Comment:
We can experiment w/ raising this; when I tested @rmfitzpatrick's inventory script (ansible/tower#816 (comment)), about 25 celery workers seemed to be a sweet spot here for leaving some breathing room on a 4GB VM.